scheduler/pkg/scheduler/scheduler.go

```diff
@@ -131,57 +134,87 @@ func (s *SimpleScheduler) scheduleToServer(modelName string) error {
 		if latestModel.HasServer() {
 			server = latestModel.Server()
 		}
-		logger.Debugf("Model %s is deleted ensuring removed", modelName)
+
+		logger.Debug("Ensuring deleted model is removed")
 		err = s.store.UpdateLoadedModels(modelName, latestModel.GetVersion(), server, []*store.ServerReplica{})
 		if err != nil {
-			logger.WithError(err).Warnf("Failed to unschedule model replicas for model %s on server %s", modelName, server)
+			logger.WithError(err).WithField("server", server).Warn("Failed to unschedule model replicas from server")
 		}
 
-	} else {
-		var debugTrail []string
-		var filteredServers []*store.ServerSnapshot
-		// Get all servers
-		servers, err := s.store.GetServers(false, true)
-		if err != nil {
-			return err
-		}
-		// Filter and sort servers
-		filteredServers, debugTrail = s.filterServers(latestModel, servers, debugTrail)
-		s.sortServers(latestModel, filteredServers)
-		ok := false
-		logger.Debugf("Model %s with desired replicas %d candidate servers %v", modelName, latestModel.DesiredReplicas(), filteredServers)
-		// For each server filter and sort replicas and attempt schedule if enough replicas
-		for _, candidateServer := range filteredServers {
-			logger.Debugf("Candidate server %s", candidateServer.Name)
-			var candidateReplicas *sorters.CandidateServer
-
-			// we need a lock here, we could have many goroutines at sorting
-			// without the store being reflected and hence sorting on stale values
-			s.muSortAndUpdate.Lock()
-			candidateReplicas, debugTrail = s.filterReplicas(latestModel, candidateServer, debugTrail)
-			if len(candidateReplicas.ChosenReplicas) < latestModel.DesiredReplicas() {
-				s.muSortAndUpdate.Unlock()
-				continue
-			}
-			s.sortReplicas(candidateReplicas)
-			err = s.store.UpdateLoadedModels(modelName, latestModel.GetVersion(), candidateServer.Name, candidateReplicas.ChosenReplicas[0:latestModel.DesiredReplicas()])
-			s.muSortAndUpdate.Unlock()
+		return nil
+	}
 
-			if err != nil {
-				logger.Warnf("Failed to update model replicas for model %s on server %s", modelName, candidateServer.Name)
-			} else {
-				ok = true
-				break
-			}
+	// Model needs to be (re)scheduled
+	var filteredServers []*store.ServerSnapshot
+
+	// Get all servers
+	servers, err := s.store.GetServers(false, true)
+	if err != nil {
+		return err
+	}
+
+	// Filter and sort servers
+	filteredServers = s.filterServers(latestModel, servers)
+	if len(filteredServers) == 0 {
+		logger.Debug("Failed to schedule model as no matching servers are available")
+		msg := fmt.Sprintf("Failed to schedule model %s as no matching servers are available", modelName)
```

Comment:
:question: :wrench: This message could probably avoid being duplicated by not including the model name.  This message is _probably_ just used to update the status for the model itself, thus the name should be irrelevant.  It is worth checking this, but it's avoid a bit of verbosity and duplication.

Reply:
Changed in https://github.com/SeldonIO/seldon-core/pull/5107/commits/f0a39c4eb8d02ae8c70fdceef08ca7a4b3b5b6c1